homeassistant/components/generic_thermostat/climate.py

```diff
@@ -521,6 +523,9 @@ async def _async_heater_turn_off(self):
 
     async def async_set_preset_mode(self, preset_mode: str):
         """Set new preset mode."""
+        preset_mode = preset_mode.lower()
```

Comment:
We shouldn't do this. If someone passes an invalid preset, we should raise, not try to match it up.

Reply:
I was hoping to be able to control both this thermostat and my ecobee's with the same presets. Looking into ecobee code they will match 'away' and 'Away', even though the modes they report are those returned by the ecobee api:
```
preset_modes:
  - Away
  - Home
  - Sleep
```
Looking at the other climate impls they all use the lowercase presets, so probably better to enforce it being lower case.